PyInstaller/loader/pyimod03_importers.py

```diff
@@ -491,6 +490,123 @@ def get_filename(self, fullname):
                 fullname.replace('.', pyi_os_path.os_sep) + '.pyc')
         return filename
 
+    def find_spec(self, fullname, path=None, target=None):
+        """
+        PEP-451 finder.find_spec() method for the ``sys.meta_path`` hook.
+
+        fullname     fully qualified name of the module
+        path         None for a top-level module, or package.__path__ for submodules or subpackages.
+
+        Finders must return ModuleSpec objects when find_spec() is called. This new method replaces find_module()
+        and find_loader() (in the PathEntryFinder case). If a loader does not have find_spec(), find_module() and
+        find_loader() are used instead, for backward-compatibility.
+        """
+        # Acquire the interpreter's import lock for the current thread. This
+        # lock should be used by import hooks to ensure thread-safety when
+        # importing modules.
+
+        filename = None  # None means - no module found in this importer.
+
+        if fullname in self.toc:
+            # Tell the import machinery to use self.load_module() to load the module.
+            filename = fullname
+            trace("import %s # PyInstaller PYZ", fullname)
+        elif path is not None:
+            # Try to handle module.__path__ modifications by the modules themselves
```

Comment:
It's not worth it because `find_module` and `load_module` are deprecated anyway. These methods will be removed after Python 2.7 support is dropped, and until that happens, Python 2.7 is in bugfix-only mode so there shouldn't be any new problems that occur with those methods.